atm.js

```diff
@@ -1,3 +1,113 @@
 $(document).ready(function(){
-// Uh oh -- it's saying `$` is undefined! Something's missing from `index.html`...
+  $("#checking .deposit").on("click", function(evt){
+    evt.preventDefault();
+    $("#checking .balance.zero").removeClass("zero");
+    $("#checking .input").css("color", "black");
+    $("#checking .input").select();
+    var totalBalance = currentBalance("#checking .balance");
+    var userInput = currentInput("#checking .input");
+    var balance = userInput + totalBalance;
+    $("#checking .balance").html("$" + balance);
+  })
+
+  $("#checking .withdraw").on("click", function(evt){
+    evt.preventDefault();
+    $("#checking .balance.zero").removeClass("zero");
+    $("#checking .input").css("color", "black");
+    $("#checking .input").select();
+    var totalBalance = currentBalance("#checking .balance");
+    var userInput = currentInput("#checking .input");
+    var balance = withdraw(totalBalance, userInput, "#checking .balance", "#checking .input")
+    $("#checking .balance").html("$" + balance);
+  })
+
+  $("#savings .deposit").on("click", function(evt){
+    evt.preventDefault();
+    $("#savings .balance.zero").removeClass("zero");
+    $("#savings .input").css("color", "black");
+    $("#savings .input").select();
+    var totalBalance = currentBalance("#savings .balance");
+    var userInput = currentInput("#savings .input");
+    var balance = userInput + totalBalance;
+    $("#savings .balance").html("$" + balance);
+  })
+
+  $("#savings .withdraw").on("click", function(evt){
+    evt.preventDefault();
+    $("#savings .input").css("color", "black");
+    $("#savings .input").select();
+    var totalBalance = currentBalance("#savings .balance");
+    var userInput = currentInput("#savings .input");
+    var balance = withdraw(totalBalance, userInput, "#savings .balance", "#savings .input");
+    $("#savings .balance").html("$" + balance);
+  })
+
+  function redTextWarning(location){
+    $(location).css("color", "red");
+    $(location).val("INSUFFICIENT FUNDS!");
+    $(location).select();
+  }
+
+  function currentBalance(location){
+    var displayedBalance = $(location).html();
+    var totalBalance = parseFloat(displayedBalance.split("$")[1]);
+    return totalBalance;
+  }
+
+  function currentInput(location){
+    var amount = $(location).val();
+    var floatAmount = parseFloat(amount).toFixed(2);
+    var userInput = parseFloat(floatAmount);
+    return userInput;
+  }
+
+  function zeroDollars(location){
+    $(location).addClass("zero");
+  }
+
```

Comment:
really like this abstraction. How can we do this same sort of thing in the 4 event listeners for the main apps functionality?